src/test/utils_wrap_thread.cpp

```diff
@@ -0,0 +1,58 @@
+#include "gtest/gtest.h"
+
+#include <vector>
+#include <atomic>
+
+#include "../utils/wrap_thread.hpp"
+#include "../libs0.hpp"
+
+TEST(utils_wrap_thread, thread_at_throws_in_middle_of_move) {
+	g_dbg_level_set(10,"test");
+
+	std::vector<wrap_thread> vec;
+	vec.resize(1);
+	vec.at(0) = std::move(wrap_thread());
```

Comment:
this is not needed, so please explain:
// yes, here we create an empty thread and immediately replace it with another one